delphi-checks/src/main/java/au/com/integradev/delphi/checks/PointerNameCheck.java

```diff
@@ -41,20 +53,28 @@ public DelphiCheckContext visit(TypeDeclarationNode type, DelphiCheckContext con
     return super.visit(type, context);
   }
 
-  private static boolean isViolation(TypeDeclarationNode type) {
+  private boolean isViolation(TypeDeclarationNode type) {
     if (type.isPointer()) {
```

Comment:
Now that there's more logic in this method, there's a lot of quite indented code - I wonder if it could be a good idea to change this `type.isPointer()` check to be an early return.